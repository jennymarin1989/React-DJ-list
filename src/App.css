.homepage {

    text-align: center;
    font-family: monospace;
    padding-top: 8em;
    font-size: 19px;
    background-image: url('./music.png');
    background-attachment: fixed;
    background-position: center; 
    background-size: 90%;

}

.songs {
    
    font-weight: bold;
    font-size: 21px;
    border: darkgray;
}

.form-control[type=text] {
        width: 30%;
        padding: 12px 20px;
        margin: 8px 0;
        display: inline-block;
        border: 2px solid rgb(14, 12, 12);
        border-radius: 4px;
        box-sizing: border-box;
        font-size: 15px;
}