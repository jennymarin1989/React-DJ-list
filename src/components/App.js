import React, { Component} from 'react';

class App extends Component {
    
  constructor(){
    super();
    this.state = {
      input: ""
    }
  }


  render(){  
    return (
      <div>
        <h1>REACT DJ</h1>
      </div> 

    )
  }
    
};



export default App;