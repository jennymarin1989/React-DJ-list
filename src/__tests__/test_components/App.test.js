import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../../components/App.js'


describe('App', () =>{

  let app = shallow(<App />)
  it("initializes correctly", () =>{
    expect(app).toMatchSnapshot();
  });

  describe('initializes state successfully', ()=> {
    let app = shallow(<App />)
    it("initializes state as an empty string", () => {
      expect(app.state('input')).toEqual("");
    });
  })

  describe('renders input correctly', () => {
    let app = shallow(<App />)
    it('renders year input correctly', () => {
      expect(app.find('#year-input').length).toEqual(1);
    })
  })

  describe('year input', () => {
    it('should responde to change event and change the state of the input', () => {
      let handleInputSpy = sinon.spy(App.prototype, "handleInputChange");
      let event = {target: { name:"input", value:"2007"}}
      let app = mount(<App />);
      app.find('#year-input').simulate('change', event )
      expect(handleInputSpy.calledOnce).toEqual(true);
    });
  })

})
