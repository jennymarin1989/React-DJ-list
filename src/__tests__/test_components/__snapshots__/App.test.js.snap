// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App initializes correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          REACT DJ
</h1>,
        <div
          className="input-group"
>
          <input
                    className="form-control"
                    id="year-input"
                    name="input"
                    onChange={undefined}
                    placeholder="Enter a year"
                    type="text"
                    value={undefined}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "REACT DJ",
        },
        "ref": null,
        "rendered": "REACT DJ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control"
            id="year-input"
            name="input"
            onChange={undefined}
            placeholder="Enter a year"
            type="text"
            value={undefined}
/>,
          "className": "input-group",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "id": "year-input",
            "name": "input",
            "onChange": undefined,
            "placeholder": "Enter a year",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            REACT DJ
</h1>,
          <div
            className="input-group"
>
            <input
                        className="form-control"
                        id="year-input"
                        name="input"
                        onChange={undefined}
                        placeholder="Enter a year"
                        type="text"
                        value={undefined}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "REACT DJ",
          },
          "ref": null,
          "rendered": "REACT DJ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              id="year-input"
              name="input"
              onChange={undefined}
              placeholder="Enter a year"
              type="text"
              value={undefined}
/>,
            "className": "input-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "year-input",
              "name": "input",
              "onChange": undefined,
              "placeholder": "Enter a year",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
